### DISCUSSION TITLE
Utilty types for Api

### DISCUSSION THREAD

**Discussion by QzCurious** [ecyrbe/zodios#216] [Ideas]
Excited to see new utilty types coming from [v10.5.0](https://github.com/ecyrbe/zodios/releases/tag/v10.5.0).

What about utilty types for api? 
- With help of `ZodiosErrorForApi`, [zodios plugins](https://www.zodios.org/docs/client/plugins#write-your-own-plugin) can handle common errors, like promping "unknown error" 500, pretty easily.
- A practice I have met is backend adds a custom http header `version` for there api. With `ZodiosResponseForApi`, [zodios plugins](https://www.zodios.org/docs/client/plugins#write-your-own-plugin) can be made to upgrade/migrate client size database (like [zustand persist](https://github.com/pmndrs/zustand#persist-middleware)).

**Comment by ecyrbe** [comment]
Hello,

Could you share what is your expected input / output of these utility types ?

**Reply by QzCurious** [reply]
## ZodiosErrorForApi

I hope it's called like `ZodiosErrorForApi<typeof apiClient.api, 500>` and return a union of all endpoint. Like how `makeApi` assembling all endpoints. And maybe default`Status` to `number | 'default'`.

I always make a `baseError` function something like:
```ts
interface BaseErrorParams<T extends z.ZodTypeAny> {
  status: number | 'default';
  errorSchema: T;
}

export const baseError = <T extends z.ZodTypeAny>(params: BaseErrorParams<T>) => {
  return {
    status: params.status,
    schema: z.object({
      error: params.errorSchema,
    }),
  } as const;
};
```
And share the same sturcture to every endpoint.
With `ZodiosErrorForApi`, I can easily target any error defined by `errorSchema` within zodios plugin.
